nixos/modules/services/matrix/lk-jwt-service.nix

```
@@ -0,0 +1,87 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.lk-jwt-service;
+in
+{
+  meta.maintainers = [ lib.maintainers.quadradical ];
+  options.services.lk-jwt-service = {
+    enable = lib.mkEnableOption "Enable lk-jwt-service";
+    package = lib.mkPackageOption pkgs "lk-jwt-service" { };
+
+    livekitUrl = lib.mkOption {
+      type = lib.types.strMatching "^wss?://.*";
+      example = "wss://example.com/livekit/sfu";
+      description = ''
+        The public websocket URL for livekit.
+        The proto needs to be either  `wss://` (recommended) or `ws://` (insecure).
+      '';
+    };
+
+    keyFile = lib.mkOption {
+      type = lib.types.path;
+      description = ''
+        Path to your LiveKit key file, with syntax `APIkey: secret`.
+        For more information, see <https://github.com/element-hq/lk-jwt-service#configuration>.
+      '';
```

The syntax part here is also unclear. This requires the same keyfile as generated for Livekit itself, correct?

---

Yes, correct.

---

Since lk-jwt-service will most likely be deployed next to livekit often, we could do something like

```nix
  default = if config.services.livekit.enable then
    config.services.livekit.keyFile
  else
    null;

---

Not sure the previous suggestion makes sense, if the livekit key file can hold multiple name/secret pairs.

---

And the key file here would be

```
lk-jwt-service: duKefEuPv5FKQVq73JXjJin1CmXEbD0G
```

---

> Since lk-jwt-service will most likely be deployed next to livekit often, we could do something like
> 
> ```nix
>   default = config.services.livekit.keyFile;
>   defaultText = literalExpression "config.services.livekit.keyFile";
> ```

It depends, since you can use a remote livekit deployment too. And yes, the livekit key can have multiple pairs, I think.

---

No strong opinions, we can also not make that assumption and leave it like that.

But I do want better documentation for the key setup, this is confusing at best.

```suggestion
      description = ''
        Path to a file containing the credentials mapping (`<keyname>: <secret>`) to access LiveKit.
        
        Example:
        ```
        lk-jwt-service: CorrectHorseBatteryStaple
        ```       
        
        For more information, see <https://github.com/element-hq/lk-jwt-service#configuration>.
      '';
```

---

Done.